lib/shoryuken/manager.rb

```diff
@@ -15,6 +15,8 @@ def initialize(fetcher, polling_strategy)
       @fetcher = fetcher
       @polling_strategy = polling_strategy
 
+      @heartbeat = Concurrent::TimerTask.new(run_now: true, execution_interval: 1, timeout_interval: 60) { dispatch }
```

Comment:
Actually, the impl above would a require a mutex wrapping the entire `dispatch`.